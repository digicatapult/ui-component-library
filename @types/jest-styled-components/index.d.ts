declare module 'jest-styled-components/serializer'
