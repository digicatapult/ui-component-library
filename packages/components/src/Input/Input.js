import React from 'react'
import { PropTypes } from 'prop-types'
import styled from 'styled-components'

import colors from '../colors'

const { errorRed, grey } = colors

const StyledInput = styled.input`
  border: 1px solid ${grey};
  border-radius: 5px;
  height: 30px;
  padding: 5px;
  min-width: 350px;
`

const Input = props => <StyledInput {...props} />

Input.defaultProps = {
  type: 'text',
  monetary: false,
  error: ''
}

Input.propTypes = {
  id: PropTypes.string.isRequired,
  error: PropTypes.string,
  type: PropTypes.string
}

export default Input
