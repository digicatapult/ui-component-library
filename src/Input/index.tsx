import React from 'react'
import styled from 'styled-components'

export interface InputTextProps {
  styles?: React.CSSProperties
  width?: string
  type: 'hidden' | 'text'
  name: string
}

const InputText: React.FC<InputTextProps> = (props) => <Input {...props} />

const Input = styled('input')<InputTextProps>`
  width: ${({ width }) => width || '200px'};
  height: 35px;
  margin-bottom: 5px;
  text-align: center;
  border-radius: 2px;
  border: 1px solid var(--black, #000);
  background: var(--white, #fff);
`

export default InputText
