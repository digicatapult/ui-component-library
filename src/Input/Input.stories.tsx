import React from 'react'

import Input, { InputTextProps } from './index.js'

export default {
  title: 'Components/HelpContainer',
  component: Input,
  argTypes: {
    name: {
      control: 'text',
    },
    type: {
      options: ['text', 'hidden'],
      control: { type: 'radio' },
    },
  },
}

const DefaultStoryTemplate = (args: InputTextProps) => <Input {...args} />

export const Default = DefaultStoryTemplate.bind({})

Default.args = {
  name: 'default',
  type: 'text',
}
