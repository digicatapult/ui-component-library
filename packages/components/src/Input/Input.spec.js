import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Input from '.'

describe('Input', () => {
  test('default', () => {
    const tree = renderer.create(<Input label="default" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('monetary', () => {
    const tree = renderer
      .create(<Input label="money related" monetary />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
