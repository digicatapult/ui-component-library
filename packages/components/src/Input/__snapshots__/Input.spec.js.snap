// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input default 1`] = `
.c0 {
  border: 1px solid #D1D1D1;
  border-radius: 5px;
  height: 30px;
  padding: 5px;
  min-width: 350px;
}

<input
  className="c0"
  id="default"
  label="default"
  type="text"
/>
`;

exports[`Input monetary 1`] = `
.c0 {
  border: 1px solid #D1D1D1;
  border-radius: 5px;
  height: 30px;
  padding: 5px;
  min-width: 350px;
}

<input
  className="c0"
  id="value"
  label="money related"
  type="text"
/>
`;
